### DISCUSSION TITLE
Using DecapCMS without Git or Oauth?

### DISCUSSION THREAD

**Discussion by webketje** [decaporg/decap-cms#7052] [Q&A]
DecapCMS looks promising, but from the docs it looks as if it cannot function without Git or an authentication mechanism?
For a lot of developers, all we desire is a local CMS front-end to separate content editing from website development, a bit like [Publii](https://getpublii.com/) or [VS Code Frontmatter](https://frontmatter.codes/). These 2 clients have serious drawbacks though as Publii cannot be "composed" with SSGs like Decap can, and VS Code Frontmatter always requires VS Code.

What I thought I was looking for is a `local-fs-gateway` backend, and I would give a shot at creating one but there's no general interface defined and I need to know whether that would even work out. Basically I have my SSG but do not wish for DecapCMS to own authentication and repository updates. All I need it to do is write to the local filesystem, and perhaps trigger a custom-defined command on save.

Another potential use case (hypothetical but realistic): I make small websites for local businesses, and they don't care that their content is git-versioned and they don't want to bother managing it either. They are fine with a button that says "backup" that will simply do a `cp -r` of their website contents on the local disk. For auth I choose to use Basic HTTP Authentication (I know, it's suboptimal but I'm just cutting corners here to be fast) and thus don't need any Decap Oauth

**Comment by mmkal** [comment]
Do these steps not work for you? https://decapcms.org/docs/beta-features/#working-with-a-local-git-repository

That should give you a "local" backend, i.e. just read and write files on disk.

**Comment by webketje** [comment]
@mmkal that looks like it, but I guess part of the question is why *must* Git be part of the workflow, why not have a separate `local-fs` backend instead of a `git-gateway` + `local_backend: true`?

> I do not wish for DecapCMS to own authentication and repository updates. All I need it to do is write to the local filesystem, and perhaps trigger a custom-defined command on save.

**Reply by privatemaker** [reply]
@webketje when i first started using Decap I struggled to understand the same things you appear to be questioning.

> ...why must Git be part of the workflow...

Git is one of the whole points of static sites / JAM stack- one gets granular versioning of content- which is quite different from a basic CRUD stored in a database or just writing to disk.

- What happens when someone accidentally makes a mistake and deletes something? 
- What happens when an edit is made, but you want prior version instead?

> All I need it to do is write to the local filesystem

This is exactly what happens using [Local git repository](https://decapcms.org/docs/working-with-a-local-git-repository/) feature, the `npx decap-server` process writes files to disk, but **does not** make git commits.

However, this is *really* only meant for local machines or where highly trusted users have access and IS NOT on the public internet. This basically allows writing to disk from an HTTP call with ZERO authentication.

> ...and perhaps trigger a custom-defined command on save.

I have never use the [CMS.registerEventListener()](https://decapcms.org/docs/registering-events/) feature yet, but I assume you can use that for some actions on saving.

**Reply by webketje** [reply]
@privatemaker I am fully aware of the merits of using Git.
I am just disappointed that the Decap UI front-end cannot be used without a git-gateway of some sort.
Ultimately I will have to set up a POC and play around with it before I reach a conclusion.

Before the JAMstack trend in JS, there were already flat-file PHP CMS'es like [Grav](https://getgrav.org/) or [GetSimpleCMS](https://get-simple.info) that can store auth credentials in text files (encrypted, salted, and protected by server software). Yes it's not as secure as a git-provider with 2FA, but it's sufficient for a small local business.

Those flat-file CMS'es can also be used with Git, for example to version the layouts, config & data files, but allow unversioned on-disk content & media editing. I don't want the local plumber or restaurant to be permanently tethered to a Git repo I manage.

**Comment by sethbrasile** [comment]
@webketje I was just in the exact same situation and ran across this! I have not tried it yet. It looks like it may answer the question for me at least? We'll see.

https://www.reddit.com/r/codestitch/comments/1e4xl00/for_the_folks_who_use_decapcms_here_i_made_a/

**Reply by sethbrasile** [reply]
It works great.

I don't see a self hostable option yet, and I also don't see any pricing... so I don't think it's technically ready for prime time for clients unless you're ok with potentially needing to migrate at some point.. I think I'll leave a comment asking about this on the reddit thread.

**Comment by kerbymart** [comment]
@webketje we use Decap, and this is a serious issue. An option to build from local or S3 would be beneficial. Not everything needs to be "git-based."

**Comment by eerison** [comment]
Hello guys, 

Just a question,  using local git. Is it possible to commit and push to remote repository from the UI. 

Or will it just create/update local files?
